lib/ukcpuid/cpuid.c

```diff
@@ -0,0 +1,182 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include <uk/errptr.h>
+#include <uk/cpuid.h>
+
+#include "../plat/common/include/x86/cpu_defs.h"
```

Comment:
Please don't include any platform-internal headers. Those aren't exposed to libraries. Make sure that you have required definitions available in your library.

Reply:
Got it. I will copy paste definitions in the new private header

Reply:
@skuenzer , I see your concern, but I don't think we want to have duplicated code either. Maybe we could define a "public" header file with required definitions and include that **both** in the platform-internal headers and the `ukcpuid` library header?